src/syrupy/extensions/single_file.py

```diff
@@ -70,4 +70,4 @@ def _write_snapshot_fossil(self, *, snapshot_fossil: "SnapshotFossil") -> None:
     def __clean_filename(self, filename: str) -> str:
         filename = str(filename).strip().replace(" ", "_")
         max_filename_length = 255 - len(self._file_extension or "")
-        return re.sub(r"(?u)[^-\w.]", "", filename)[:max_filename_length]
+        return re.sub(r"(?u)[^-\w.\[\]]", "", filename)[:max_filename_length]
```

Comment:
Likely a breaking change, since snapshot files would get renamed

Reply:
If we are going to do this then it's probably worth supporting the full range of characters which user can narrow down if they choose to override

Reply:
What's the overlap between characters that can appear in a test name and characters supported by all three operating systems we support?

Reply:
I'm thinking we set the list as any characters except `\/:*"<>|` `NUL` and control characters.

Reply:
I do wonder if a whitelist is more reliable.

Also if we expand the character set, arguably we shouldn't replace whitespace with underscores, since whitespace is allowed on Windows/Mac/Linux. Personally, I like the "simpler" characters b/c they're easier to work with in the terminal. Something like whitesapce requires you to escape the char. There's a trade off here I suppose.

Reply:
I too like the simple charset. I'm inclined to leave it as is for now until there's pressure to support more characters by default... no need to introduce a breaking change at the moment. I can just create an issue to track this, we should probably resolve it before v1 though